In [46]:
import pandas as pd

In [11]:
text1 = "Ethics and rules are important in running any business. It is the foundation of humans and their systems "

In [12]:
len(text1)

105

In [13]:
text2 = text1.split(' ')

In [14]:
text2

['Ethics',
 'and',
 'rules',
 'are',
 'important',
 'in',
 'running',
 'any',
 'business.',
 'It',
 'is',
 'the',
 'foundation',
 'of',
 'humans',
 'and',
 'their',
 'systems',
 '']

Finding words with length grater than 3

In [15]:
[w for w in text2 if len(w)>4]

['Ethics',
 'rules',
 'important',
 'running',
 'business.',
 'foundation',
 'humans',
 'their',
 'systems']

Finding capitalized words


In [16]:
[w for w in text2 if w.istitle()]

['Ethics', 'It']

In [17]:
[w for w in text2 if w.endswith('s')]

['Ethics', 'rules', 'is', 'humans', 'systems']

Finding unique words

In [18]:
text3 = "To be or not to be."

In [21]:
text4 = text3.split()
text4

['To', 'be', 'or', 'not', 'to', 'be.']

In [22]:
len(set(text4))

6

In [23]:
len(set([w.lower() for w in text4]))

5

# Cleaning text

In [24]:
text5 = "She sells see shells on the sea shore."

In [25]:
text6 = text5.strip()

In [28]:
text6.split(' ')

['She', 'sells', 'see', 'shells', 'on', 'the', 'sea', 'shore.']

# Handling larger text

In [47]:
f = open('train.dat','r')

Read the first line.

In [48]:
f.readline()

'sentiment\tText\n'

In [49]:
f.readline()

"2\tshe had another song out recently but it didn't get that much play\n"

Reset the reading pointer

In [50]:
f.seek(0)

0

Reading the full file.

In [51]:
sdata = f.read()

In [52]:
len(sdata)

2173922

2173922 this the number of characters in the file


In [53]:
sdata = sdata.splitlines()

In [54]:
len(sdata)

31001

31001 is the number of lines separated by \n


In [55]:
sdata[7]

"2\tdon't you procrastinate!   (like I do...)"

regex with pandas

In [58]:
df = pd.DataFrame(sdata,columns=['text'])

In [60]:
df.head()

text
0                                    sentiment\tText
1  2\tshe had another song out recently but it di...
2  2\twellity! i think ima gonna clean my room :|...
3  2\tBug in damn cod4 system link. My map pack i...
4  8\tweeooow, i feel fat.... i remember when i u...

number of characters in top 5 rows 

In [62]:
df['text'].head().str.len()

0     14
1     68
2    128
3     56
4    122
Name: text, dtype: int64

find number of token for each element of series

In [64]:
df['text'].head().str.split().str.len()

0     2
1    14
2    27
3    12
4    19
Name: text, dtype: int64

find which entries contain a particular word

In [65]:
df['text'].head().str.contains('my')

0    False
1    False
2     True
3    False
4    False
Name: text, dtype: bool

find all occurances of a character

In [75]:
df['text'].head().str.findall(r'\d')

0        []
1       [2]
2       [2]
3    [2, 4]
4       [8]
Name: text, dtype: object

In [76]:
df


text
0                                        sentiment\tText
1      2\tshe had another song out recently but it di...
2      2\twellity! i think ima gonna clean my room :|...
3      2\tBug in damn cod4 system link. My map pack i...
4      8\tweeooow, i feel fat.... i remember when i u...
5      7\tmmmm now you're speaking my language! (Unfo...
6                             7\tReally?? I feel special
7           2\tdon't you procrastinate!   (like I do...)
8      1\tIve been wanting 2 but will be gone 4 2 mo....
9      2\tSushi two nights in a row isn't THAT wrong ...
10       2\thaha i get my senior license next friday!!!!
11     7\tUMMMNNN ummmnnn good thanks for letting me ...
12                            3\tNobody is up this early
13     2\tCan't stop coughing, feel like I might have...
14     3\tI also have to make space on my computer fo...
15     3\targ..tried one pic, but she moved so i fail...
16     2\tMigraining today--the naproxen sodium table...
17     3\tStorysize is 2 (and it's posted as a short ...
18                      1\tLooks like my Twitter is down
19     10\tOh yeah the camera clipping problems with ...
20     9\tBeing in Vegas with @donniewahlberg would b...
21     4\theylo johnn (: im a huge fan. hope ur day i...
22                  3\tis coming back to Ruston tomorrow
23     2\tI caaaaan't sleep.....it's 3.30am! wahhhh.....
24     2\tJust found out I have to leave my lovely fl...
25     7\tACK! Ghost Whisperer isn't on!   Damn netwo...
26     4\thaha, happened to me a lot of times. Know t...
27     2\tits not fruit, its those sugar covered swee...
28     2\tPoor Jon, I feel for you dude   Eat some ch...
29     2\tHearing a surprising news about my bestfrie...
...                                                  ...
30971                            2\ti miss music on mtv!
30972  1\tJenn jenn jenn oh why oh why won't u pass o...
30973  8\tMake her something. Buying something is too...
30974                           4\t- Such a pretty baby!
30975  1\tOh poor thing  Hope it's easily sorted out....
30976  8\tShe must have a very adwancrd sense of humo...
30977  3\tis taking Ethan and Ryder to playgroup this...
30978             1\tsooo long twitters! I'm off to Cuba
30979                           8\toh that was good cake
30980  10\tthat is my idea of good dinner  happy munchin
30981     9\twell thats not nice... hope ur all good now
30982  8\tSeeing my brother graduate tomorrow. Should...
30983  2\tshack idol today(: can't wait. it's gonna b...
30984  10\tnice recolouring of your page  nobody is b...
30985                    10\tthank you. yes yes. hooray!
30986  4\tHa. J perfect timing on that message, the c...
30987                  1\tmy knees are sore after physio
30988                                      3\t- awesome.
30989                          2\tOff the road this week
30990                              4\thappy mother's day
30991  8\tIs one bar two of the pub crawl... Gonn be ...
30992  3\tI would like to be able to go just to ONE o...
30993              3\tIt's freakin' hot and humid today.
30994  10\tI know how you feel  Have a lovely relaxin...
30995          10\tGood show, sleepytime. Goodnight moon
30996                             3\tRainy day is rainy.
30997                       2\twhat was it? I missed it.
30998  3\tIf the drive is automatically ejecting supp...
30999           8\tmmmm... looks so yummy! good seeing u
31000                                6\tF�licitations!

[31001 rows x 1 columns]